tests/model_registry/utils.py

```diff
@@ -156,3 +161,71 @@ def get_model_registry_db_label_dict(db_resource_name: str) -> dict[str, str]:
         Annotations.KubernetesIo.INSTANCE: db_resource_name,
         Annotations.KubernetesIo.PART_OF: db_resource_name,
     }
+
+
+def get_pod_container_error_status(pod: Pod) -> str:
+    pod_instance_status = pod.instance.status
+    # Check the containerStatuses and if any containers is in waiting state, return that information:
+
+    for container_status in pod_instance_status.get("containerStatuses", []):
+        if waiting_container := container_status.get("state", {}).get("waiting"):
+            return waiting_container["reason"] if waiting_container.get("reason") else waiting_container
+        return ""
+    return ""
+
+
+def get_not_running_pods(pods: list[Pod]) -> list[Any]:
+    pods_not_running = []
+    try:
+        for pod in pods:
+            pod_instance = pod.instance
+            if container_status_error := get_pod_container_error_status(pod=pod):
+                pods_not_running.append({pod.name: container_status_error})
+            # Waits for all pods in a given namespace to be in final healthy state(running/completed).
+            # We also need to keep track of pods marked for deletion as not running. This would ensure any
+            # pod that was spin'ed up in place of pod marked for deletion, reaches healthy state before end
+            # of this check
```

Comment:
please use proper docstring to explain what the function is for

Reply:
The comment is clearly stating what the function does. With mypy, doc string for args and return value is redundant.

Reply:
a user should not need to dive into the function code to understand what it does. please move to a docsting

Reply:
ok, moving this comment to the beginning of the function.